reactor-core/src/main/java/reactor/core/publisher/SinksSpecs.java

```diff
@@ -81,16 +77,23 @@ public final Emission tryEmitComplete() {
 		if (done) {
 			return Sinks.Emission.FAIL_TERMINATED;
 		}
+		Thread lockedAt = this.lockedAt;
+		if (!(lockedAt == null || lockedAt == Thread.currentThread())) {
+			return Emission.FAIL_NON_SERIALIZED;
```

Comment:
we'll need to document that fail fast behavior everywhere we pretend that `Sinks` are by default safe to use from multiple threads (mainly in various places of `Sinks.java` javadocs)